### PR TITLE
Initial version of QSP feature.

### PR DESCRIPTION
**Pull request by cgranade**
This PR adds a new package, Microsoft.Quantum.Research.Simulation, with an initial version of the quantum signal processing feature for use with simulation functionality in the core Quantum Development Kit. Thanks to Jeongwan Haah for taking leadership in getting this feature out there!

### CODE REVIEW

**Comment by rmshaffer** [comment]
Is the code in the F# files under `simulation/qsp` brand new code, or is it derived from somewhere else? What is the scope/level of granularity of code review you'd like for those files?

File: build/manifest.ps1 [hunk 1]
```diff
@@ -11,6 +11,7 @@
     );
     Assemblies = @(
         ".\src\characterization\bin\$Env:BUILD_CONFIGURATION\netstandard2.1\Microsoft.Quantum.Research.Characterization.dll",
-        ".\src\chemistry\bin\$Env:BUILD_CONFIGURATION\netstandard2.1\Microsoft.Quantum.Research.Chemistry.dll"
+        ".\src\chemistry\bin\$Env:BUILD_CONFIGURATION\netstandard2.1\Microsoft.Quantum.Research.Chemistry.dll",
+        ".\src\simulation\qsp\bin\$Env:BUILD_CONFIGURATION\netstandard2.1\Microsoft.Quantum.Research.Simulation.Qsp.dll"
```
Comment (efratshabtai) [inline build/manifest.ps1:15]: I wonder if this should be Microsoft.Research.Simulation.dll that will have in the future all simulation related code, not just QSP. If this assembly meant to be QSP only then why don't we name it Microsoft.Research.QSP?
The package name is Microsoft.Research.Simulation - do we expect it to have multiple dlls?
What about other packages like Microsoft.Resaerch.Chemistry?

File: src/characterization/characterization.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
     <PackageProjectUrl>https://github.com/Microsoft/Quantum-NC</PackageProjectUrl>
     <PackageIconUrl>https://secure.gravatar.com/avatar/bd1f02955b2853ba0a3b1cdc2434e8ec.png</PackageIconUrl>
     <PackageTags>Quantum Q# Qsharp</PackageTags>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
```
Comment (efratshabtai) [inline src/characterization/characterization.csproj:17]: What does that mean in terms of user experience?
Will they need to accept a license when building the code? running it?
How will that look like?

Comment (cgranade) [reply src/characterization/characterization.csproj:17]: At the moment (see https://github.com/NuGet/Home/issues/8299 for discussion), this is only a hint to the VS 2019 package management GUI that indicates a user may want to look at the license when first installing the package into a project.

**Pull request merged**